test/Microsoft.AspNet.Mvc.ModelBinding.Test/Metadata/ModelBinderAttributeTest.cs

```diff
@@ -25,5 +25,47 @@ public void InvalidBinderType_Throws()
             // Assert
             Assert.Equal(expected, ex.Message);
         }
+
+        [Fact]
+        public void NoBinderType_NoBindingSource()
+        {
+            // Arrange
+            var attribute = new ModelBinderAttribute();
+
+            // Act
+            var source = attribute.BindingSource;
+
+            // Assert
+            Assert.Null(source);
+        }
+
+        [Fact]
+        public void BinderType_CustomBindingSource()
+        {
+            // Arrange
+            var attribute = new ModelBinderAttribute();
+            attribute.BinderType = typeof(ByteArrayModelBinder);
+
+            // Act
+            var source = attribute.BindingSource;
+
+            // Assert
+            Assert.Equal(BindingSource.Custom, source);
+        }
+
+        [Fact]
+        public void BinderType_OverrideCustomBindingSource()
```

Comment:
Changed to to `BinderType_SettingBindingSource_OverridesDefaultCustomBindingSource()`

By default if the BinderType is set then the source will be set to custom. This tests that you can override that